src/plugins/gui/util/initializeProviders.ts

```diff
@@ -14,6 +14,21 @@ export async function initializeProviders<T>(providerFactories: Array<FiatProvid
   const providerPromises: Array<Promise<FiatProvider<T>>> = []
 
   const getTokenIdProvider = (pluginId: string, currencyCode: string) => getTokenId(account.currencyConfig[pluginId], currencyCode)
+  const getTokenIdFromContract = (params: { pluginId: string; contractAddress: string }) => {
+    const { pluginId, contractAddress } = params
+    const currencyInfo = account.currencyConfig[pluginId]
+    const edgeToken = Object.entries(currencyInfo.allTokens).find(([tokenId, token]) => {
+      const { networkLocation } = token
+      if (networkLocation == null) return false
+      if (networkLocation.contractAddress == null) return false
+      if (networkLocation.contractAddress.toLowerCase() === contractAddress.toLowerCase()) return true
+      return false
+    })
+    if (edgeToken == null) return
+
+    // Return the tokenId which is the first entry in the tuple
+    return edgeToken[0]
```

Comment:
You can use `Object.values` instead of `Object.entries` to avoid the accessor (`[0]`).

Reply:
Nope, the tokenId is the key, not the value.

Reply:
Ah.